test/buffer.jl

```diff
@@ -0,0 +1,70 @@
+using Serialization: deserialize, serialize
+using Test
+using ray_core_worker_julia_jll: LocalMemoryBuffer, Data, Size, OwnsData, IsPlasmaBuffer
+
+@testset "LocalMemoryBuffer" begin
+    data = UInt16[1:3;]
+
+    @testset "non-copied object reference" begin
+        buffer = LocalMemoryBuffer(pointer_from_objref(data), sizeof(data), false)
+        @test Data(buffer[]) == pointer_from_objref(data)
+        @test Size(buffer[]) == sizeof(data)
+        @test !OwnsData(buffer[])
```

Comment:
what are the semantics of this?  set on construction (if teh constructor copies the data into its own memory)?

ah yeah, here we go: https://github.com/ray-project/ray/blob/a03efd9931128d387649dd48b0e4864b43d3bfb4/src/ray/common/buffer.h#L69-L70